### ISSUE TITLE
Remove Bundle, Bundle Builder types

### ISSUE THREAD

**Issue by Kashomon** [GoogleCloudPlatform/k8s-cluster-bundle#183]
An idea: how about removing the bundle and the bundle builder? Would simplify things.

Then, you just build components individually. The CLI has the ability to take a list of components. It can even output a ComponentSet (even version it if you like).

**Comment by Kashomon** [comment]
For example: `bundlectl build --files=comp-builder1.yaml,comp-builder2.yaml` 

One trickiness: a nice thing about the current approach is that it's easy to chain.

`bundlectl build foo.yaml | bundlectl patch | bundlectl filter | bundlectl export`

**Comment by Kashomon** [comment]
I suppose the documents could still be coalesced into a single document via '---'

**Comment by Kashomon** [comment]
@johnbelamaric @justinsb @johnsonj -- thoughts?

**Comment by Kashomon** [comment]
I've convinced myself this should probably happen. Probably for the v0.10 release

**Comment by Kashomon** [comment]
One problem I just realized is that we added a build option to the Bundle Builder: https://github.com/Kashomon/k8s-cluster-bundle/blob/master/pkg/apis/bundle/v1alpha1/builder_types.go#L42  

```go
	// ComponentNamePolicy defines how to generate the metadata.name
	// for a Component or ComponentBuilder that does not already have one.
	//  - SetAndComponent generates a name from the set name and version and
	//    component name and version.
	//  - Component (default) generates a name from the component name and version
	ComponentNamePolicy string `json:"componentNamePolicy,omitempty"`
```

**Comment by johnbelamaric** [comment]
I am currently using BundleBuilder to gather up a bunch of components, and then use ExportAsObjects to get a ComponentSet + all Components, which I then export to YAML (as CRDs).

I don't strictly need these types. But I do need a way to operate on a whole component set (e.g., what BundleWrapper does now when you BuildAllPatchTemplates on a Bundle).

**Comment by Kashomon** [comment]
Good point. Maybe the BundleBuilder stays but the Bundle itself goes away.
